crate/findex/src/batcher_findex.rs

```diff
@@ -0,0 +1,299 @@
+use std::{collections::HashSet, fmt::Debug, hash::Hash, sync::Arc};
+
+use cosmian_sse_memories::{ADDRESS_LENGTH, Address, BatchingMemoryADT, MemoryBatcher};
+
+use crate::{Decoder, Encoder, Findex, IndexADT, adt::IndexBatcher, error::BatchFindexError};
+
+#[derive(Debug)]
+pub struct FindexBatcher<
+    const WORD_LENGTH: usize,
+    Value: Send + Hash + Eq,
+    EncodingError: Send + Debug,
+    BatcherMemory: Clone + Send + BatchingMemoryADT<Address = Address<ADDRESS_LENGTH>, Word = [u8; WORD_LENGTH]>,
+> {
+    memory: BatcherMemory,
+    encode: Encoder<Value, BatcherMemory::Word, EncodingError>,
+    decode: Decoder<Value, BatcherMemory::Word, EncodingError>,
+}
+
+impl<
+    const WORD_LENGTH: usize,
+    Value: Send + Hash + Eq,
+    BatcherMemory: Debug
+        + Send
+        + Sync
+        + Clone
+        + BatchingMemoryADT<Address = Address<ADDRESS_LENGTH>, Word = [u8; WORD_LENGTH]>,
+    EncodingError: Send + Debug,
+> FindexBatcher<WORD_LENGTH, Value, EncodingError, BatcherMemory>
+{
+    pub fn new(
+        memory: BatcherMemory,
+        encode: Encoder<Value, BatcherMemory::Word, EncodingError>,
+        decode: Decoder<Value, BatcherMemory::Word, EncodingError>,
+    ) -> Self {
+        Self {
+            memory,
+            encode,
+            decode,
+        }
+    }
+
+    // Both insert and delete operations make an unbounded number of calls to
+    // `guarded_write` on the memory layer.
```

Comment:
I am not sure why you need this comment here. The Findex paper analyses the performance of Findex operations and your comment just rewords a consequence of the lock-free property.